man/cidcall.1
.\" %W% %G%
.TH CIDCALL 1
.SH NAME
.B cidcall
- view calls, hangups, and messages the NCID call file
.SH SYNOPSIS
.B cidcall
[-r | -chmo] [cidlog]
.SH DESCRIPTION
The
.B cidcall
scrip displays the cidcall.log file.
.SS "OPTIONS"
.PD 0
.TP 14
.B -r
Display the raw call file.
.br
Default: format the call file.
.TP
.B -c
Display CID lines (incoming calls) in the call file.
.TP
.B -h
Display HUP lines (terminated calls) in the call file.
.TP
.B -m
Display MSG lines (messages) in the call file.
.TP
.B -o
Display OUT lines (outgoing calls) in the call file.
.PD
.SS "Arguments"
.PD 0
.TP 14
.B cidlog
The NCID call file.
.br
Default: /var/log/cidcall.log
.PD
.SH SEE ALSO
ncidd.conf.5,
ncidd.alias.5,
ncidd.blacklist.5,
cidalias.1,
cidupdate.1


doc/CHANGES
ncid - Network CallerID

Version 0.85
    - README: updated
    - Makefile: updated
    - renamed cidgate/ to gateway/
    - client/Makefile: updated
    - client/ncid.sh:
        added verbose levels 1-9
        added doVerbose{}
        changed all "if $Verbose" to "doVerbose <string> 1"
        added test for $ProgDir == "" to correctly set $Program from $ProgName
        updated comments for $ProgDir and $ProgName
        added --alt-date|-A option to display date as dd/mm/yyyy
        added country code for HR (Croatia)
        fixed "call" array pronlem that crashed ncid if call
            not found in the array.
        added date separator ($DateSepar) so "-" or "." can be used
            in place of "/" in displayed date
        changed --message option to do nothing, messages always sent
            to the output module
        removed -M | --messages option from Usage:
        added code to set ring indirectly from a module
            name with "-" replaced by "_" in ncid.conf
        getCID{}: removed verbose inside "if $Raw {"
                  added line: if {$Program != ""} {doVerbose "$dataBlock" 1}
    - client/ncid.conf-in:
            added AltDate configuration variable
            added ring number configuration
    - debian/README.Debian: updated
    - debian/Makefile: updated
    - debian/ncid-*.init: updated
    - debian/ncid-notify.init: new
    - debian/postrm-client: updated
    - debian/prerm-client: updated
    - debian/rules: updated
    - doc/INSTALL-andLinux: removed
    - doc/INSTALL-Cygwin: added rebase and service documentation
    - doc/INSTALL-Fedora: updated
    - doc/INSTALL-Mac: added section to install using launchctl
    - doc/INSTALL-TiVo: updated for the startncid script.
    - doc/INSTALL-Ubuntu: added section to install from GetDeb
    - doc/INSTALL-Win: updated
    - doc/Verbose-ncidd: updated
    - doc/NCID-SDK.odt: updated
    - Fedora/Makefile: updated
    - Fedora/ncid.spec: updated
    - Fedora/ncid-notify.init: new
    - Fedora/ncid-notify.service-in: new
    - Fedora/ncidd.service-in: changed network.target to multi-user.target
    - Fedora/ncid2ncid.service-in: changed network.target to multi-user.target
    - Fedora/ncid2sip.service-in: changed network.target to multi-user.target
    - Fedora/ncid2yac.service-in: changed network.target to multi-user.target
    - FreeBSD/Makefile: updated
    - FreeBSD/ncid-notify.rc: new
    - gateway/Makefile: updated
    - gateway/sip2ncid.c:
        added print of command line options to logfile
        doPCAP(): - changed  errorExit() call
        getOptions(): - added "--osx-launchd" for Mac launchd
        parseLines: fixed linenum[0] never populated
    - gateway/sip2ncid.h: updated Usage: 
    - gateway/ncid2ncid.c:
        added "OUT: " & "HUP: " to lineType[]
       added print of command line options to logfile
       processPackets(): moved if (outcall) to inside above if statement
       getOptions(): - added "--osx-launchd" for Mac launchd
    - gateway/ncid2ncid.h: updated Usage: 
    - Mac/*: new
    - man/Makefile: updated
    - man/ncid.1: updated
    - man/notify.1: new
    - man/ncid.conf.5: updated
    - man/ncid2ncid.1: updated
    - man/ncidd.8: updated
    - man/ncidd.conf.5: Updated
    - man/ncidd.blacklist.5: updated
    - man/ncidd.whitelist.5: new
    - man/ncidmodules.1: removed
    - man/ncidmodules.conf.5: removed
    - man/ncid-modules.7: new
    - man/ncid-tivo.1: moved to TiVo/
    - man/ncid-*.1: rewrote
    - mancid*.1: rewrote
    - man/ncidrotate.1: moved to scripts/
    - man/ncidtoold.1: removed
    - man/sip2ncid.conf.5: Updated
    - modules/Makefile.modules: updated
    - modules/README.modules: updated
    - modules/ncid-initmodem.sh: modified comments
    - modules/ncid-kpopup.sh: modified comments
    - modules/ncid-mythtv.sh: modified comments
    - modules/ncid-notify.sh: new
    - modules/ncid-tivo.sh: moved to TiVo/
    - modules/*.sh: lots of code minor changes in all modules
    - modules/ncidmodules.conf-in: removed
    - modules/ncid-*.conf: new
    - scripts/README.scripts: updated
    - scripts/README.scripts: renamed to README.logrotate
    - scripts/Makefile: updated
    - scripts/ncidrotate.1: moved from man/
    - scripts/ncid.logrotate: updated to work with all service files & systemctl
    - scripts/ncidrotate.sh: improved code a bit
    - renamed scripts/ to logrotate/
    - server/Makefile: updated
    - server/README.server: updated
    - server/ncidd.blacklist: added 4 area codes to block
    - server/ncidd.whitelist: new
    - server/ncidd.c:
        changed ringline to infoline, initialize to ONELINE instead of "-"
        Updated CIDINFO line comment for sendInfo(int mainsock)
        main() - added: strncpy(infoline, lineid, CIDSIZE - 1);
        removed extern void builtinAlias(), userAlias();
        Moved position of command line printout in logfile
        formatCID() - changed callout to calltype with values IN, OUT, and HUP
                    - removed builtinAlias() & userAlias()
                    - added test for "CALLHUP" in the CALL line from a gateway
                    - added switch statement for linelabel value
        main(): - added call whitelist lines under if (hangup)
        getOptions(): - added whitelist option
                      - added "--osx-launchd" for Mac launchd
        reload(): - moved blacklist reload inside if (hangup)
                  - added whitelist reload
        main(): - added code to print total alias entries
    - server/ncidd.conf-in: added whitelist variable
    - server/ncidd.h: - added defines for IN, OUT, and HUP
                      - added whitelist option to USAGE define
                      - replaced HANGUPMSG define with BLMSG and WLMSG
                      - updated Usage:
    - server/nciddalias.c:idoAlias(): - added Alias Entries log message
                           - added builtinAlias() & userAlias() was in ncidd.c 

tools/cidcall.pl
#!/usr/bin/perl

# cidcall - caller ID report

# Created by John L. Chmielewski on Fri Sep 14, 2001
# modified by jlc on Tue Jul 30, 2002
#    - modified to look for lines that start with CID or EXTRA
#    - modified to locate NUMBER or NMBR field
#    - modified $date to work with MMDD and MMDDYYYY data
# modified by Aron Green on Sun Nov 24, 2002
#    fixed regular expression to handle "'" in the name field
# modified by jlc on Mon Apr 7, 2003
#    fixed regular expression to handle 11 digit numbers
# modified by jlc on Fri Aug 12, 2005
#    modified to display the new "LINE" filed
# modified by jlc on Fri Apr 14, 2006
#    added -m option, modified raw option to print all lines, changed name
# modified by jlc on Wed Mar 28, 2007
#    - modified regular expressions to look for * with key word
# modified by jlc on Sun May 16, 2010
#    - removed <> from enclosing line label,
#    - changed number from (xxx)xxx-xxxx to xxx-xxx-xxxx
# modified by jlc on Sun Feb 13, 2011
#    - added -cho options
#    - the -chmo options can be combined
#    - formats and prints the entire call log file by default
#    - prints the line labels when formatted

use Getopt::Std;

getopts('chmor') || die "Usage: cidcall [-r | -chom] [cidlog]\n";

($cidlog = shift) || ($cidlog = "/var/log/cidcall.log");

format STDOUT =
@<<< @<<<<<<<<<<<<<<< @<<<<<<<<<<<<< @<<<<<<<<<< @<<<<< @<<<<<<<<<
$label, $name,           $number,       $date,      $time, $line
.

open(CIDLOG, $cidlog) || die "Could not open $cidlog\n";

while (<CIDLOG>) {
  if (!$opt_r && !$opt_c && !$opt_h && !$opt_m && !$opt_o) {&parseLine;}
  elsif ($opt_r) {print;}
  else {
    if ($opt_c) { if (/CID:/) {&parseLine;} }
    if ($opt_h) { if (/HUP:/) {&parseLine;} }
    if ($opt_m) { if (/MSG:/) {print;} }
    if ($opt_o) { if (/OUT:/) {&parseLine;} }
  }
}

sub parseLine {
    ($label, $date, $time, $number, $name) = 
     /(\w+:).*\*DATE.(\d+).*\*TIME.(\d+).*\*NU*MBE*R.([-\w\s]+).*\*NAME.(.*)\*+$/;
    ($line) = /.*\*LINE.([-\w\d]+).*/;
    $line =~ s/-*//;
    $date =~ s/(\d\d)(\d\d)(\d\d\d\d)*/$1\/$2\/$3/;
    $date =~ s/\/$//;
    $time =~ s/(\d\d)(\d\d)/$1:$2/;
    $number =~ s/\d?(\d\d\d)(\d\d\d)(\d\d\d\d)/$1-$2-$3/;
     write;
}
